### DISCUSSION TITLE
Finding antiviral names

### DISCUSSION THREAD

**Discussion by jgonzalezm** [msberends/AMR#75] [Q&A] [answered]
as.ab () able to identify antivirals and antifungals?

I am having trouble with antivirals being identified as another medication.  It seems like it's searching for antivirals in the antifungal table.

For example:

> ab_from_text("bictegravir")
[[1]]
Class <ab>
[1] NYS

> is.ab("bictegravir")
[1] FALSE

I would expect

**Comment by msberends** [comment]
Hi! Thanks for using our package. That’s true, currently there is a `antivirals` data set, but `as.ab()` only searches for antibiotics and antifungals in the `antibiotics` data set. Perhaps it’s time for v2.0 to combine them and facilitate your use case. 

Perhaps it’s time for `as.av()` and `av_from_text()`. We’ll track this in #76. Thanks for the great idea!

**Comment by msberends** [comment, marked as answer]
This issue brought extensive support for antivirals to our package!

You can test the beta version using:

```r
install.packages("remotes") # if you haven't already
remotes::install_github("msberends/AMR")
```

## How to use the new antivirals support

Use `as.av()` to validate drugs as antivirals:

```r
as.av("bictegravir")
#> Class 'av'
#> [1] EMT+TEA+BIC

av_name("bictegravir")
#> [1] "Emtricitabine/tenofovir alafenamide/bictegravir"
```

We took the drug names from the WHOCC website, and it seems that  bictegravir is missing as a name on its own. Is this true? We could add it manually. For now, it’s only available as combination with emtri and teno-ala.

Some more examples [from our website](https://msberends.github.io/AMR/reference/as.av.html):

```r
# these examples all return "ACI", the ID of aciclovir:
as.av("J05AB01")
#> Class 'av'
#> [1] ACI
as.av("J 05 AB 01")
#> Class 'av'
#> [1] ACI
as.av("Aciclovir")
#> Class 'av'
#> [1] ACI
as.av("aciclo")
#> Class 'av'
#> [1] ACI
as.av("   aciclo 123")
#> Class 'av'
#> [1] ACI
as.av("ACICL")
#> Class 'av'
#> [1] ACI
as.av("ACI")
#> Class 'av'
#> [1] ACI
as.av("Virorax") # trade name
#> Class 'av'
#> [1] ACI
as.av("Zovirax") # trade name
#> Class 'av'
#> [1] ACI

as.av("acyklofir") # severe spelling error, yet works
#> Class 'av'
#> [1] ACI

# use av_* functions to get a specific properties (see ?av_property);
# they use as.av() internally:
av_name("J05AB01")
#> [1] "Aciclovir"
av_name("acicl")
#> [1] "Aciclovir"
```

**Reply by jgonzalezm** [reply]
The as.av functionality looks great.   Thank you very much for incorporating it into the package so quickly.

 The drugs I've tested so far worked.  

A follow-up question though:  Should is.av ("bictegravir) return TRUE?

I am getting the following.

> as.av("bictegravir")
Class 'av'
[1] EMT+TEA+BIC
> is.av("bictegravir") # would expect TRUE
[1] FALSE
> is.ab("atorvastatin") # Expect False
[1] FALSE
> is.ab("dolutegravir") # Expect False
[1] FALSE
> is.av("dolutegravir") #Expect True
[1] FALSE
> as.ab("dolutegravir") # not an antiviral
Class 'ab'
[1] <NA>
Warning message:
in as.ab(): these values could not be coerced to a valid antimicrobial ID:
"dolutegravir". 
> as.av("dolutegravir")
Class 'av'
[1] DOL
